mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/chess/rules/MovesTest.kt

```diff
@@ -144,4 +147,275 @@ class MovesTest {
+    assertThat(actions.asIterable()).isEmpty()
+  }
+
+  @Test
+  fun castling_adversaryKing() {
+    val board =
+        buildBoardWithHistory<Piece<Role>> {
+          yield(
+              buildBoard {
+                set(Position(4, 7), adversaryKing)
+                set(Position(7, 7), alliedRook)
+              },
+          )
+        }
+    val actions = board.rightCastling().map { it.first }
+    assertThat(actions.asIterable()).isEmpty()
+  }
+
+  @Test
+  fun castling_adversaryRook() {
+    val board =
+        buildBoardWithHistory<Piece<Role>> {
+          yield(
+              buildBoard {
+                set(Position(4, 7), alliedKing)
+                set(Position(7, 7), adversaryRook)
+              },
+          )
+        }
+    val actions = board.rightCastling().map { it.first }
+    assertThat(actions.asIterable()).isEmpty()
+  }
+
+  @Test
+  fun castling_alliedButNotKing() {
+    val board =
+        buildBoardWithHistory<Piece<Role>> {
+          yield(
+              buildBoard {
+                set(Position(4, 7), alliedPawn)
+                set(Position(7, 7), alliedRook)
+              },
+          )
+        }
+    val actions = board.rightCastling().map { it.first }
+    assertThat(actions.asIterable()).isEmpty()
+  }
+
+  @Test
+  fun castling_alliedButNotRook() {
+    val board =
+        buildBoardWithHistory<Piece<Role>> {
+          yield(
+              buildBoard {
+                set(Position(4, 7), alliedKing)
+                set(Position(7, 7), alliedPawn)
+              },
+          )
+        }
+    val actions = board.rightCastling().map { it.first }
+    assertThat(actions.asIterable()).isEmpty()
+  }
+
+  @Test
+  fun enPassant_emptyHistory_isSuccessful() {
+    val board =
+        buildBoardWithHistory<Piece<Role>> {
+          yield(
+              buildBoard {
+                set(Position(0, 3), alliedPawn)
+                set(Position(1, 1), adversaryPawn)
+              },
+          )
+          yield(
+              buildBoard {
+                set(Position(0, 3), alliedPawn)
+                set(Position(1, 3), adversaryPawn)
+              },
+          )
+        }
+    val actions = board.enPassant(Position(0, 3), Delta(1, 0)).map { it.first }
+    assertThat(actions.asIterable()).containsExactly(Action(Position(0, 3), Delta(1, -1)))
+  }
+
+  @Test
+  fun enPassant_inBetweenMove() {
+    val board =
+        buildBoardWithHistory<Piece<Role>> {
+          yield(
+              buildBoard {
+                set(Position(0, 3), alliedPawn)
+                set(Position(1, 2), adversaryPawn)
+              },
+          )
+          yield(
+              buildBoard {
+                set(Position(0, 3), alliedPawn)
+                set(Position(1, 3), adversaryPawn)
+              },
+          )
+        }
+    val actions = board.enPassant(Position(0, 3), Delta(1, 0)).map { it.first }
+    assertThat(actions.asIterable()).isEmpty()
+  }
+
+  @Test
+  fun enPassant_badRow() {
```

Comment:
Suggestion: Test name should reflect expected outcome
```suggestion
  fun enPassant_badRow_isNotAllowed() {
```